torchvision/prototype/transforms/functional/_meta.py

```diff
@@ -184,7 +184,31 @@ def _gray_to_rgb(grayscale: torch.Tensor) -> torch.Tensor:
     return grayscale.repeat(repeats)
 
 
-_rgb_to_gray = _FT.rgb_to_grayscale
+def rgb_to_grayscale(image: torch.Tensor, num_output_channels: int = 1) -> torch.Tensor:
```

Comment:
This is actually deprecated:

https://github.com/pytorch/vision/blob/7a62a545ce76f43ccc5cfe0009131f7db14ae7b5/torchvision/prototype/transforms/functional/_deprecated.py#L25

Can't we merge the `c == 3` branch into `convert_color_space_image_tensor`? If there is actually a perf implication and we need the faster op as standalone, I'm ok to have a `_rgb_to_gray` as before that only does this and can be used from the other color kernels.

Reply:
I do not follow you about deprecated method. Deprecated method is inside `_deprecated.py`, here I replaced `_rgb_to_gray = _FT.rgb_to_grayscale`. If there is something to do with `convert_color_space_image_tensor`, I'd prefer to keep it for another PR.

Reply:
You defined a function `rgb_to_grayscale` here. There are several issues with this:

1. It clashes with the function of the same name in the deprecated namespace.
2. It doesn't adhere to the naming convention with `features.ColorSpace.GRAY` since it uses the term `grayscale`
3. It reintroduces the whole concept of `num_output_channels` which we also deprecated. `convert_color_space` no longer has this concept.

Basically what I'm asking here is that you implement your perf improvements into `_rgb_to_gray`. Since that is private we can also remove all of the error checking. Basically the whole function should look like

```py
def _rgb_to_gray(image: torch.Tensor) -> torch.Tensor:
    r, g, b = image.unbind(dim=-3)
    return (r * 0.2989).add_(g, alpha=0.587).add_(b, alpha=0.114)
```